### DISCUSSION TITLE
SSH authentication for git push to a PR branch?

### DISCUSSION THREAD

**Discussion by anjohnson** [community/community#24018] [New to GitHub]
<p>I’ve been reviewing a pull request to a project I maintain, and wanted to add a commit to the PR. I have hidden the people, project and branch names below since those details shouldn’t matter. Towards the bottom of the PR’s page it says</p>
<blockquote>
<p>Add more commits by pushing to the <code>branch-name</code> branch on <strong>reporter-name/project-name</strong></p>
</blockquote>
<p>I’ve been using SSH for all my git interactions with GitHub for years, so I ran this command to do just that:</p>
<pre><code class="lang-auto">$ git push git://github.com/reporter-name/project-name.git reporter-name-branch-name:branch-name
fatal: remote error: 
  You can't push to git://github.com/reporter-name/project-name.git
  Use https://github.com/reporter-name/project-name.git
</code></pre>
<p>Having to switch to https with a password seemed strange, but it did work (once I’d looked up my password). However I immediately received a Deprecation Notice email from The GitHub Team:</p>
<blockquote>
<p>You recently used a password to access the repository at reporter-name/project-name with git using git/2.29.1.</p>
<p>Basic authentication using a password to Git is deprecated and will soon no longer work. Visit <a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" class="inline-onebox" rel="noopener nofollow ugc">Token authentication requirements for Git operations - The GitHub Blog</a> for more information around suggested workarounds and removal dates.</p>
</blockquote>
<p>The blog post above implies I can use an SSH key for Git all authentication with GitHub, but I wasn’t able to do that to push to my reporter’s branch. Is that ability just not in place yet, did I find someplace they missed, did I make a mistake somehow, or is this particular case not going to be supported at all?</p>

**Comment by byrneh** [comment, marked as answer]
<p><a class="mention" href="/anjohnson">@anjohnson</a>, welcome to the community<br>
Your SSH remote syntax looks incorrect.</p>
<p>Does it work if you use<br>
<a href="mailto:git@github.com">git@github.com</a>:USERNAME/REPOSITORY?</p>

**Comment by anjohnson** [comment]
<blockquote>
<p>Your SSH remote syntax looks incorrect.</p>
</blockquote>
<p>That is wrong isn’t it, yet I copied it from the PR’s command line instructions “Checkout via command line” section with the Git option selected which I assumed would be the same as SSH but evidently isn’t, maybe that’s what needs fixing. I thought selecting that used to update the URLs in the full list of instructions but it doesn’t seem to now; the repo’s Clone pull-down does gives the right syntax for “SSH”, and doesn’t have a “Git” option at all. The recommendation in the “remote error:” message from my <code>git push</code> could also have provided better guidance.</p>
<p>Okay, thanks <a class="mention" href="/byrneh">@byrneh</a> – my mistake, although GitHub could make some improvements to help people like me avoid making it in the future.</p>

**Comment by byrneh** [comment]
<aside class="quote no-group" data-username="anjohnson" data-post="1" data-topic="164652">
<div class="title">
<div class="quote-controls"></div>
 anjohnson:</div>
<blockquote>
<pre><code class="lang-auto">You can't push to git://github.com/reporter-name/project-name.git
  Use https://github.com/reporter-name/project-name.git
</code></pre>
</blockquote>
</aside>
<p>Hi <a class="mention" href="/anjohnson">@anjohnson</a>, most importantly you are up and working with SSH again  which is good news :slight_smile:</p>
<p>I will admit it confused me when I first looked at your post.  Then looking at your post for a while, eureka I realised that it was not saying you cannot use SSH and must use  HTTPS, the logic thought your had mistyped your HTTPS syntax as the URL (: afterwards) you entered was HTTP syntax other that you had ‘<strong>git</strong>’ before the :  rather than ‘<strong>https</strong>’.</p>
<p>Which in your case was incorrect. If the error message had given both SSH and git examples formats you would have known your error at once.</p>
<p>There is good probability the error message is from your Git client rather than GitHub server. But that is not something I know for a fact one way or the other.<br>
Thanks for marking post as a solution, as this helps others when searching and looking for similar problems and solutions.</p>
